File: src/main/java/com/teamwizardry/librarianlib/fx/shader/LibShaders.java
package com.teamwizardry.librarianlib.fx.shader;

import com.teamwizardry.librarianlib.fx.shader.uniforms.FloatTypes;

public enum LibShaders {
	INSTANCE;
	
	public static Shader HUE;
	
	LibShaders() {
		initShaders();
	}
	
	private void initShaders() {
		HUE = new HueShader(null, "/assets/librarianlib/shaders/hue.frag");
		ShaderHelper.addShader(HUE);
	}
	
	public static class HueShader extends Shader {

		public FloatTypes.Float hue;

		public HueShader(String vert, String frag) {
			super(vert, frag);
		}
		
		@Override
		public void initUniforms() {
			hue = getUniform("hue");
		}

		@Override
		public void uniformDefaults() {
			hue.set(0);
		}
		
	}
}

File: src/main/java/com/teamwizardry/librarianlib/fx/shader/LibShaders.kt
package com.teamwizardry.librarianlib.fx.shader

import com.teamwizardry.librarianlib.fx.shader.uniforms.FloatTypes

object LibShaders {
    @JvmStatic
    val HUE = HueShader(null, "/assets/librarianlib/shaders/hue.frag")

    init {
        ShaderHelper.addShader(HUE)
    }

    class HueShader(vert: String?, frag: String?) : Shader(vert, frag) {

        var hue: FloatTypes.Float1? = null

        override fun initUniforms() {
            hue = getUniform<FloatTypes.Float1>("hue")
        }

        override fun uniformDefaults() {
            hue?.set(0f)
        }

    }
}
